### DISCUSSION TITLE
take profit and stop loss behavior

### DISCUSSION THREAD

**Discussion by nkami** [kernc/backtesting.py#242] [Q&A] [answered]
Hey, I was wondering what happens if you set both take profit and stop loss on an order, and the next candle fluctuates in a manner that should trigger both. What gets priority according to the backtest? (given the fact we cant know the order of events within the candle). 
For example: you set a market order with tp=2 and sl=1, and the next candle has high=3 and low=0.

**Comment by kernc** [comment, marked as answer]
If you enter with a plain market order, SL has priority (the framework takes an adversarial, rather than an optimistic stance).

If you enter with a stop/limit order, a warning is emitted:
https://github.com/kernc/backtesting.py/blob/a49122c72b8e0dd9b30104957638640653c2c113/backtesting/backtesting.py#L905-L915

**Reply by nkami** [reply]
Thank you!
